client/mm/mm_arb_market_maker.go

```diff
@@ -0,0 +1,653 @@
+		UseMaxFeeRate: true,
+		UseSafeTxSize: true,
+	})
+	if err != nil {
+		return fmt.Errorf("failed to get fees: %v", err)
+	}
+
+	sellSwapFees, sellRedeemFees, sellRefundFees, err := a.core.SingleLotFees(&core.SingleLotFeesForm{
+		Host:          a.host,
+		Base:          a.base,
+		Quote:         a.quote,
+		UseMaxFeeRate: true,
+		UseSafeTxSize: true,
+		Sell:          true,
+	})
+	if err != nil {
+		return fmt.Errorf("failed to get fees: %v", err)
+	}
+
+	buyFundingFees, err := a.core.MaxFundingFees(a.quote, a.host, uint32(len(a.cfg.BuyPlacements)), a.cfg.QuoteOptions)
+	if err != nil {
+		return fmt.Errorf("failed to get funding fees: %v", err)
+	}
+
+	sellFundingFees, err := a.core.MaxFundingFees(a.base, a.host, uint32(len(a.cfg.SellPlacements)), a.cfg.BaseOptions)
+	if err != nil {
+		return fmt.Errorf("failed to get funding fees: %v", err)
+	}
+
+	a.feesMtx.Lock()
+	defer a.feesMtx.Unlock()
+
+	a.buyFees = &orderFees{
+		swap:       buySwapFees,
+		redemption: buyRedeemFees,
+		funding:    buyFundingFees,
+		refund:     buyRefundFees,
+	}
+	a.sellFees = &orderFees{
+		swap:       sellSwapFees,
+		redemption: sellRedeemFees,
+		funding:    sellFundingFees,
+		refund:     sellRefundFees,
+	}
+
+	return nil
+}
+
+func (a *arbMarketMaker) run() {
+	book, bookFeed, err := a.core.SyncBook(a.host, a.base, a.quote)
+	if err != nil {
+		a.log.Errorf("Failed to sync book: %v", err)
+		return
+	}
+	a.book = book
+
+	a.updateFeeRates()
+
+	err = a.cex.SubscribeMarket(a.ctx, dex.BipIDSymbol(a.base), dex.BipIDSymbol(a.quote))
+	if err != nil {
+		a.log.Errorf("Failed to subscribe to cex market: %v", err)
+		return
+	}
+
+	tradeUpdates, unsubscribe, tradeUpdatesID := a.cex.SubscribeTradeUpdates()
+	defer unsubscribe()
+	a.cexTradeUpdatesID = tradeUpdatesID
+
+	wg := &sync.WaitGroup{}
+
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+		for {
+			select {
+			case <-bookFeed.Next():
+				// Really nothing to do with the updates. We just need to keep
+				// the subscription live in order to get VWAP on dex orderbook.
+			case <-a.ctx.Done():
+				return
+			}
+		}
+	}()
+
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+		for {
+			select {
+			case update := <-tradeUpdates:
+				a.handleCEXTradeUpdate(update)
+			case <-a.ctx.Done():
+				return
+			}
+		}
+	}()
+
+	noteFeed := a.core.NotificationFeed()
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+		defer noteFeed.ReturnFeed()
+		for {
+			select {
+			case n := <-noteFeed.C:
+				a.handleNotification(n)
+			case <-a.ctx.Done():
+				return
+			}
+		}
+	}()
+
+	wg.Wait()
+	a.cancelAllOrders()
+}
+
+func RunArbMarketMaker(ctx context.Context, cfg *BotConfig, c clientCore, cex libxc.CEX, log dex.Logger) {
+	if cfg.ArbMarketMakerConfig == nil {
+		// implies bug in caller
+		log.Errorf("No arb market maker config provided. Exiting.")
+		return
+	}
+
+	mkt, err := c.ExchangeMarket(cfg.Host, cfg.BaseAsset, cfg.QuoteAsset)
+	if err != nil {
+		log.Errorf("Failed to get market: %v", err)
+		return
+	}
+
```

Comment:
I wonder if we need to somehow account for the ability of the exchange operator to change market parameters, e.g. max fee rate or lot size.

Reply:
Yeah I think so. Core could send a notification regarding this. I'll make an issue.